File: notification-api/src/main/java/com/outbox/notification_api/service/NotifyService.java
package com.outbox.notification_api.service;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Service;

@Service
public class NotifyService {

    private final KafkaTemplate<String, String> kafkaTemplate;

    @Value("${spring.kafka.topic.outbox-processed}")
    private String topic;

    public NotifyService(KafkaTemplate<String, String> kafkaTemplate) {
        this.kafkaTemplate = kafkaTemplate;
    }

    public void send(String message) {
        System.out.println(message);
        kafkaTemplate.send(topic, message);
    }

}

File: user-api/src/main/kotlin/com/outbox/user_api/service/NotifyService.kt
package com.outbox.user_api.service

import org.springframework.beans.factory.annotation.Value
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Service

@Service
class NotifyService(private val kafkaTemplate: KafkaTemplate<String,Any>) {

    @Value("\${spring.kafka.topic.user-created}")
    private lateinit var userCreatedTopic: String


    fun notify(message: String) {
        println(userCreatedTopic)
        println(message)
        kafkaTemplate.send(userCreatedTopic,message)
    }
}